tools/apigee-x-trial-provision/README.md

```diff
@@ -127,3 +127,180 @@ the `CERTIFICATES` environment variable:
 |`supplied`|User-supplied in `RUNTIME_SSL_CERT` and `RUNTIME_SSL_KEY`|$RUNTIME_HOST_ALIAS or else $ORG-eval.apigee.net|
 <!-- markdownlint-enable MD013 -->
+
+## Apigee X instance with Custom Networking Provisioning
+<!-- markdownlint-disable MD013 -->
+
+## VPC Network Planning
+
+There is always a step to plan your network with regards to CIDR allocations.
+
+GCP/GKE
+CIDR: 10.0.0.0/14 CIDR IP Range: 10.0.0.0 - 10.3.255.255
+
+CIDR: 10.0.0.0/16 CIDR IP Range: 10.0.0.0 - 10.0.255.255
+
+NOTE: GCP recommended naming convention: <https://cloud.google.com/solutions/best-practices-vpc-design>
+
+### Bastion VM in a default network
+
+As Apigee X instance provisioning is a long-running operation, we would recommend to use your working PC terminal or to provision a bastion VM. Bastion VM is also useful for troubleshooting.
+
+1. In the GCP Console, select Compute Engine/VM instances hamburger menu item.
+
+1. Press Create Instance button
+
+1. Accept defaults and press Create button at the bottom of the page
+
+1. SSH into the Bastion VM instance-1. Press the SSH button.
+
+1. Login into your account to have necessary permissions.
+
+    ```sh
+    gcloud auth login --quiet
+    ```
+
+### Custom Network and Subnet Creation
+
+1. Define PROJECT variable by the ID of your project
+
+    ```sh
+    export PROJECT=<project-id>
+    ```
+
+1. Define environment variables that describe network topology
+
+    ```sh
+    export NETWORK=exco-vpc
+    export SUBNET=exco-vpc-dev-subnet
+
+    export REGION=us-central1
+    export ZONE=us-central1
+    export AX_REGION=us-central1
+
+    export NETWORK_CIDR=10.0.0.0/14
+    export SUBNET_CIDR=10.0.0.0/16
+    ```
+
+1. GCP: Create a VPC Network
+
+    ```sh
+    gcloud compute networks create $NETWORK \
+        --subnet-mode=custom \
+        --bgp-routing-mode=regional \
+        --mtu=1460
+    ```
+
+    Output:
+
+    ```sh
+    Created [https://www.googleapis.com/compute/v1/projects/qwiklabs-gcp-01-da1a8d6ae8b2/global/networks/exco-vpc].
+    NAME      SUBNET_MODE  BGP_ROUTING_MODE  IPV4_RANGE  GATEWAY_IPV4
+    exco-vpc  CUSTOM       REGIONAL
+    ...
+    ```
+
+1. Firewall Rules for internal, 22, and icmp traffic
+
+    ```sh
+    gcloud compute firewall-rules create $NETWORK-allow-internal --network $NETWORK --allow tcp,udp,icmp --source-ranges $NETWORK_CIDR
+    ```
+
+    Output:
+
+    ```sh
+    Creating firewall...⠹Created [https://www.googleapis.com/compute/v1/projects/qwiklabs-gcp-01-da1a8d6ae8b2/global/firewalls/exco-vpc-allow-internal].
+    Creating firewall...done.                                                                                                
+    NAME                     NETWORK   DIRECTION  PRIORITY  ALLOW         DENY  DISABLED
+    exco-vpc-allow-internal  exco-vpc  INGRESS    1000      tcp,udp,icmp        False
+    ```
+
+    ```sh
+    gcloud compute firewall-rules create fr-$NETWORK-ssh --network $NETWORK --allow tcp:22
```

Comment:
is this needed and is it needed for all tags?

Reply:
without this, we are not able to connect/ssh to a vm, no?

Reply:
Yeah but can we say `--target-tags ssh-access`  or something that we don't allow ssh on all VMs but only where we need to?

Reply:
How? those are for the MIG machines we want to log into for troubleshooting?